specs/core/0_beacon-chain.md

```diff
@@ -729,205 +723,191 @@ def int_sqrt(n: int) -> int:
 
 ### On startup
 
-A valid block with slot `0` (the "genesis block") has the following values. Other validity rules (eg. requiring a signature) do not apply.
+A valid block with slot `INITIAL_SLOT_NUMBER` (a "genesis block") has the following values. Other validity rules (eg. requiring a signature) do not apply.
 
 ```python
 {
-    'slot': 0,
-    'randao_reveal': bytes32(0),
+    'slot': INITIAL_SLOT_NUMBER,
+    'randao_reveal': ZERO_HASH,
     'candidate_pow_receipt_roots': [],
-    'ancestor_hashes': [bytes32(0) for i in range(32)],
+    'ancestor_hashes': [ZERO_HASH for i in range(32)],
     'state_root': STARTUP_STATE_ROOT,
     'attestations': [],
     'specials': [],
-    'proposer_signature': [0, 0]
+    'proposer_signature': [0, 0],
 }
 ```
 
 `STARTUP_STATE_ROOT` is the root of the initial state, computed by running the following code:
 
 ```python
-def on_startup(current_validators: List[ValidatorRecord],
-               pre_fork_version: int,
-               initial_validator_entries: List[Any],
+def on_startup(initial_validator_entries: List[Any],
                genesis_time: int,
                processed_pow_receipt_root: Hash32) -> BeaconState:
-    # Induct validators
-    validators = []
-    for pubkey, deposit_size, proof_of_possession, withdrawal_credentials, \
-            randao_commitment in initial_validator_entries:
-        validators, _ = get_new_validators(
-            current_validators=validators,
+    # Activate validators
+    initial_validator_registry = []
+    for pubkey, deposit, proof_of_possession, withdrawal_credentials, randao_commitment in initial_validator_entries:
+        initial_validator_registry, _ = get_new_validators(
+            current_validators=initial_validator_registry,
             fork_data=ForkData(
-                pre_fork_version=pre_fork_version,
-                post_fork_version=pre_fork_version,
-                fork_slot_number=2**64 - 1,
+                pre_fork_version=INITIAL_FORK_VERSION,
+                post_fork_version=INITIAL_FORK_VERSION,
+                fork_slot=2**64 - 1,
```

Comment:
I'm fine with it, but I thought you would parameterize it as well. 😂

Reply:
I'm not 100% clear why we want `fork_slot=2**64 - 1` as opposed to `fork_slot=0` as the initial value. @vbuterin